src/main/java/bio/terra/service/snapshot/flight/delete/DeleteSnapshotDeleteSamGroupStep.java

```diff
@@ -38,8 +42,8 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         context
             .getWorkingMap()
             .get(SnapshotWorkingMapKeys.SNAPSHOT_AUTH_DOMAIN_GROUPS, new TypeReference<>() {});
-    // Only delete the Sam group if it matches the expected naming pattern
-    var expectedName = IamService.constructSamGroupName(snapshotId.toString());
+    // Only delete the Sam group if it matches the expected name
+    var expectedName = snapshotRequestDao.getByCreatedSnapshotId(snapshotId).samGroupName();
     if (Objects.nonNull(authDomains) && authDomains.contains(expectedName)) {
```

Comment:
Now that we know we own the Sam group and it is associated with this snapshot, I think we no longer need to check if it is assigned as the auth domain before deleting it. This really just served as a sanity check that it was the right group before we were saving the name in the database.

Reply:
This also means we can remove this logic from one of the previous steps: https://github.com/DataBiosphere/jade-data-repo/pull/1744/files#diff-76f431d85e5d9b81d3f86c2801f144cd72b58454391a62dafcd605618e2f1413R67